Cargo.toml

```diff
@@ -7,3 +7,6 @@ authors = ["Sam Phippen <samphippen@googlemail.com>"]
 tokio-core = { git = "https://github.com/tokio-rs/tokio-core/" }
 futures = "0.1"
 env_logger = "0.3.4"
+
+[profile.release]
+lto = true
```

Comment:
http://llvm.org/docs/LinkTimeOptimization.html

TL;DR: your code might get faster + smaller. It's off by default because it _significantly_ increases compile times, but in release builds many are willing to pay that cost.